primitive/message.go

```diff
@@ -133,6 +133,10 @@ func (m *Message) UnmarshalProperties(data []byte) {
 	}
 }
 
+func (m *Message) GetProperties() map[string]string {
+	return m.properties
```

Comment:
I think it would be better to use a deep copy here. Reference copy may cause multi-threaded to read and write competition.